ddtrace/internal/datadog/profiling/dd_wrapper/src/synchronized_sample_pool.cpp

```diff
@@ -0,0 +1,93 @@
+#include "synchronized_sample_pool.hpp"
+
+#include "libdatadog_helpers.hpp"
+
+extern "C"
+{
+#include "datadog/common.h"
+}
+
+namespace Datadog {
+
+void
+sample_delete_fn(void* sample)
+{
+    delete static_cast<Sample*>(sample);
+}
+
+SynchronizedSamplePool::SynchronizedSamplePool()
+{
+    size_t capacity = 4;
+    const char* capacity_env = std::getenv("DD_PROFILING_SAMPLE_POOL_CAPACITY");
+    if (capacity_env) {
+        try {
+            capacity = std::stoul(capacity_env);
+        } catch (const std::exception& e) {
+            std::cerr << "Failed to parse DD_PROFILING_SAMPLE_POOL_CAPACITY: " << e.what() << std::endl;
+        }
+        if (capacity == 0) {
+            std::cerr << "Invalid DD_PROFILING_SAMPLE_POOL_CAPACITY: " << capacity_env << std::endl;
+            capacity = 4;
+        }
+    }
+
+    ddog_ArrayQueue_NewResult array_queue_new_result = ddog_ArrayQueue_new(capacity, sample_delete_fn);
+    if (array_queue_new_result.tag == DDOG_ARRAY_QUEUE_NEW_RESULT_OK) {
+        pool = std::unique_ptr<ddog_ArrayQueue, Deleter>(array_queue_new_result.ok);
+    } else {
+        auto err = array_queue_new_result.err;
+        std::string errmsg = err_to_msg(&err, "Failed to create sample pool");
+        std::cerr << errmsg << std::endl;
+        ddog_Error_drop(&err);
+        pool = nullptr;
+    }
+}
+
+std::optional<Sample*>
+SynchronizedSamplePool::get_sample()
```

Comment:
super nit:  would it help anything if these interfaces more explicitly communicated the ownership status of the underlying data?  So using a verb like `checkout` (or `take`) instead of `get`.

(we do not follow this discipline elsewhere in the repo, we have no standards on what verbs to use in different , and the suggestion itself isn't even objectively helpful, so please take this as a suggestion and not an attempt at imposing a  requirement)